app/updatemanager.go

```diff
@@ -386,6 +409,81 @@ func (c *ControlMapPool) ClearExpired() {
 		},
 	)
 	c.Pool = newPool
+
+	c.saveToStore()
+}
+
+type controlMapPoolDbFormat struct {
+	Active  []*UpdateControlMap `json:"active"`
+	Expired []*UpdateControlMap `json:"expired"`
+}
+
+// Must only be called from functions that have already locked the mutex.
+func (c *ControlMapPool) saveToStore() {
+	if c.store == nil {
+		// Saving not enabled. This should not happen outside tests.
```

Comment:
I considered it, but decided not to. If `c.store == nil`, then the program will crash because this check does not exist in `LoadFromStore`, which is part of the normal execution flow (but not part of most tests). So we will get to know about it either way, and a log message here would only clutter the test output.